Servers/Administration/OJS.Servers.Administration/Controllers/Api/SubmissionsController.cs

```diff
@@ -20,18 +24,45 @@ public class SubmissionsController : BaseAdminApiController<
     SubmissionAdministrationServiceModel>
 {
     private readonly ISubmissionsBusinessService submissionsBusinessService;
+    private readonly IGridDataService<Submission> submissionsGridDataService;
+    private readonly ILecturerContestPrivilegesBusinessService lecturerPrivilegesBusinessService;
 
     public SubmissionsController(
         IGridDataService<Submission> submissionsGridDataService,
         ISubmissionsBusinessService submissionsBusinessService,
         SubmissionsAdministrationModelValidator validator,
-        IValidator<BaseDeleteValidationModel<int>> submissionsDeleteValidator)
+        IValidator<BaseDeleteValidationModel<int>> submissionsDeleteValidator,
+        ILecturerContestPrivilegesBusinessService lecturerPrivilegesBusinessService)
         : base(
             submissionsGridDataService,
             submissionsBusinessService,
             validator,
-            submissionsDeleteValidator) =>
+            submissionsDeleteValidator)
+    {
         this.submissionsBusinessService = submissionsBusinessService;
+        this.submissionsGridDataService = submissionsGridDataService;
+        this.lecturerPrivilegesBusinessService = lecturerPrivilegesBusinessService;
+    }
+
+    [HttpGet]
+    [ProtectedEntityAction(false)]
+    public override async Task<IActionResult> GetAll([FromQuery]PaginationRequestModel model)
+    {
+        var user = this.User.Map<UserInfoModel>();
+
```

Comment:
Does this needs to be override? Maybe we should just expose a way to pass in order by?

Reply:
Removed, because this logic is already inside the SubmissionsDataService and the filter is passed to the query automatically.